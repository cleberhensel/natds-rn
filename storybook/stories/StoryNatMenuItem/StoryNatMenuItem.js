import React from 'react';
import { Text, ScrollView, SafeAreaView } from 'react-native';
import { NatContainer, NatMenuItem } from '../../../lib';

export const StoryNatMenuItem = () => {
  const menuList = [
    {
      id: 1,
      name: 'First menu',
      level: 1,
    },
    {
      id: 2,
      name: 'Second menu',
      level: 1,
    },
    {
      id: 3,
      name: 'Another sample menu',
      level: 1,
    },
    {
      id: 4,
      name: 'Another sample menu',
      level: 1,
    },
  ];
  const menuListSvg = [
    {
      id: 1,
      name: 'outlined-navigation-home',
      svg: {
        name: 'outlined-navigation-home',
        size: '24',
      },
      level: 1,
    },
    {
      id: 2,
      name: 'outlined-content-lock',
      svg: {
        name: 'outlined-content-lock',
      },
      level: 1,
    },
    {
      id: 3,
      name: 'outlined-content-wifi',
      svg: {
        name: 'outlined-content-wifi',
      },
      level: 1,
    },
    {
      id: 4,
      name: 'outlined-content-adddocument',
      svg: {
        name: 'outlined-content-adddocument',
      },
      level: 1,
    },
    {
      id: 5,
      name: 'outlined-communication-chat',
      svg: {
        name: 'outlined-communication-chat',
      },
      level: 1,
    },
    {
      id: 6,
      name: 'outlined-content-channel',
      svg: {
        name: 'outlined-content-channel',
      },
      level: 1,
    },
    {
      id: 7,
      name: 'outlined-finance-moneypaper',
      svg: {
        name: 'outlined-finance-moneypaper',
      },
      level: 1,
    },
    {
      id: 8,
      name: 'outlined-communication-speaker',
      svg: {
        name: 'outlined-communication-speaker',
      },
      level: 1,
    },
    {
      id: 9,
      name: 'outlined-place-bus',
      svg: {
        name: 'outlined-place-bus',
      },
      level: 1,
    },
    {
      id: 10,
      name: 'outlined-alert-info',
      svg: {
        name: 'outlined-alert-info',
      },
      level: 1,
    },
    {
      id: 11,
      name: 'outlined-navigation-exit',
      svg: {
        name: 'outlined-navigation-exit',
      },
      level: 1,
    },
  ];
  const menuListIconUri = [
    {
      name: 'Menu Food',
      icon: 'https://s3-sa-east-1.amazonaws.com/somos-natura-prd/files/menus/261/img_url_tmp_261_20180323135427.png',
      level: '1',
    },
    {
      name: 'Bus Information',
      icon: 'https://s3-sa-east-1.amazonaws.com/somos-natura-prd/files/menus/generic_menu/img_url_tmp_generic_menu_20170822160140.png',
      level: '1',
    },
  ];

  return (
    <SafeAreaView>
      <NatContainer style={{ backgroundColor: '#FFF', width: 300, paddingTop: 20 }}>
        <ScrollView>
          <Text>Without Icon</Text>
          {menuList.map(item => (
            <NatMenuItem
              title={item.name}
              key={item.id}
              level={item.levell}
              onPress={() => { alert('click event'); }}
            />
          ))}
          <Text>Use SVG</Text>
          {menuListSvg.map(item => (
            <NatMenuItem
              title={item.name}
              icon={{ svg: item.svg }}
              key={item.id}
              level={item.levell}
              onPress={() => { alert('click event'); }}
            />
          ))}
          <Text>With External Icon URL</Text>
          {menuListIconUri.map(item => (
            <NatMenuItem
              title={item.name}
              icon={{ uri: item.icon }}
              key={item.name}
              level={item.levell}
              onPress={() => { alert('click event'); }}
            />
          ))}
        </ScrollView>
      </NatContainer>
    </SafeAreaView>
  );
};

StoryNatMenuItem.propTypes = {
};
