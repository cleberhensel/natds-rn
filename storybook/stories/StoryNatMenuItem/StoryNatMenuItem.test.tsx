import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components/native';
import { StoryNatMenuItem } from './StoryNatMenuItem';

describe('StoryNatMenuItem component', () => {
  it('Should render StoryNatMenuItem component ', () => {
    const button = renderer.create(<StoryNatMenuItem />).toJSON();

    expect(button).toMatchSnapshot('StoryNatMenuItem snapshot');
  });
});
