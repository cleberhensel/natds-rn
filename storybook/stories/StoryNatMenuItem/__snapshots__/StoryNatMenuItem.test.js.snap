// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StoryNatMenuItem component Should render StoryNatMenuItem component  1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#FFFFFF",
          "bottom": 0,
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
          "left": 0,
          "paddingBottom": 0,
          "paddingLeft": 16,
          "paddingRight": 16,
          "paddingTop": 0,
          "right": 0,
          "top": 0,
        },
        Object {
          "backgroundColor": "#FFF",
          "marginTop": 0,
          "width": 750,
        },
      ]
    }
  >
    <RCTScrollView>
      <View>
        <Text>
          Without Icon
        </Text>
        <View
          accessible={true}
          focusable={true}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "height": 48,
              "justifyContent": "space-between",
              "marginLeft": 16,
              "opacity": 1,
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "alignSelf": "center",
                  "color": "#333333",
                  "flexBasis": 0,
                  "flexGrow": 1,
                  "flexShrink": 1,
                  "marginLeft": 32,
                },
              ]
            }
            type="default"
          >
            First menu
          </Text>
        </View>
        <View
          accessible={true}
          focusable={true}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "height": 48,
              "justifyContent": "space-between",
              "marginLeft": 16,
              "opacity": 1,
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "alignSelf": "center",
                  "color": "#333333",
                  "flexBasis": 0,
                  "flexGrow": 1,
                  "flexShrink": 1,
                  "marginLeft": 32,
                },
              ]
            }
            type="default"
          >
            Second menu
          </Text>
        </View>
        <View
          accessible={true}
          focusable={true}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "height": 48,
              "justifyContent": "space-between",
              "marginLeft": 16,
              "opacity": 1,
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "alignSelf": "center",
                  "color": "#333333",
                  "flexBasis": 0,
                  "flexGrow": 1,
                  "flexShrink": 1,
                  "marginLeft": 32,
                },
              ]
            }
            type="default"
          >
            Another sample menu
          </Text>
        </View>
        <View
          accessible={true}
          focusable={true}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "height": 48,
              "justifyContent": "space-between",
              "marginLeft": 16,
              "opacity": 1,
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "alignSelf": "center",
                  "color": "#333333",
                  "flexBasis": 0,
                  "flexGrow": 1,
                  "flexShrink": 1,
                  "marginLeft": 32,
                },
              ]
            }
            type="default"
          >
            Another sample menu
          </Text>
        </View>
        <Text>
          Use SVG
        </Text>
        <View
          accessible={true}
          focusable={true}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "height": 48,
              "justifyContent": "space-between",
              "marginLeft": 16,
              "opacity": 1,
            }
          }
        >
          <SvgMock
            height="24"
            style={
              Object {
                "fill": "#333333",
              }
            }
            width="24"
          />
          <Text
            style={
              Array [
                Object {
                  "alignSelf": "center",
                  "color": "#333333",
                  "flexBasis": 0,
                  "flexGrow": 1,
                  "flexShrink": 1,
                  "marginLeft": 32,
                },
              ]
            }
            type="default"
          >
            outlined-navigation-home
          </Text>
        </View>
        <View
          accessible={true}
          focusable={true}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "height": 48,
              "justifyContent": "space-between",
              "marginLeft": 16,
              "opacity": 1,
            }
          }
        >
          <SvgMock
            height={24}
            style={
              Object {
                "fill": "#333333",
              }
            }
            width={24}
          />
          <Text
            style={
              Array [
                Object {
                  "alignSelf": "center",
                  "color": "#333333",
                  "flexBasis": 0,
                  "flexGrow": 1,
                  "flexShrink": 1,
                  "marginLeft": 32,
                },
              ]
            }
            type="default"
          >
            outlined-content-lock
          </Text>
        </View>
        <View
          accessible={true}
          focusable={true}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "height": 48,
              "justifyContent": "space-between",
              "marginLeft": 16,
              "opacity": 1,
            }
          }
        >
          <SvgMock
            height={24}
            style={
              Object {
                "fill": "#333333",
              }
            }
            width={24}
          />
          <Text
            style={
              Array [
                Object {
                  "alignSelf": "center",
                  "color": "#333333",
                  "flexBasis": 0,
                  "flexGrow": 1,
                  "flexShrink": 1,
                  "marginLeft": 32,
                },
              ]
            }
            type="default"
          >
            outlined-content-wifi
          </Text>
        </View>
        <View
          accessible={true}
          focusable={true}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "height": 48,
              "justifyContent": "space-between",
              "marginLeft": 16,
              "opacity": 1,
            }
          }
        >
          <SvgMock
            height={24}
            style={
              Object {
                "fill": "#333333",
              }
            }
            width={24}
          />
          <Text
            style={
              Array [
                Object {
                  "alignSelf": "center",
                  "color": "#333333",
                  "flexBasis": 0,
                  "flexGrow": 1,
                  "flexShrink": 1,
                  "marginLeft": 32,
                },
              ]
            }
            type="default"
          >
            outlined-content-adddocument
          </Text>
        </View>
        <View
          accessible={true}
          focusable={true}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "height": 48,
              "justifyContent": "space-between",
              "marginLeft": 16,
              "opacity": 1,
            }
          }
        >
          <SvgMock
            height={24}
            style={
              Object {
                "fill": "#333333",
              }
            }
            width={24}
          />
          <Text
            style={
              Array [
                Object {
                  "alignSelf": "center",
                  "color": "#333333",
                  "flexBasis": 0,
                  "flexGrow": 1,
                  "flexShrink": 1,
                  "marginLeft": 32,
                },
              ]
            }
            type="default"
          >
            outlined-communication-chat
          </Text>
        </View>
        <View
          accessible={true}
          focusable={true}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "height": 48,
              "justifyContent": "space-between",
              "marginLeft": 16,
              "opacity": 1,
            }
          }
        >
          <SvgMock
            height={24}
            style={
              Object {
                "fill": "#333333",
              }
            }
            width={24}
          />
          <Text
            style={
              Array [
                Object {
                  "alignSelf": "center",
                  "color": "#333333",
                  "flexBasis": 0,
                  "flexGrow": 1,
                  "flexShrink": 1,
                  "marginLeft": 32,
                },
              ]
            }
            type="default"
          >
            outlined-content-channel
          </Text>
        </View>
        <View
          accessible={true}
          focusable={true}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "height": 48,
              "justifyContent": "space-between",
              "marginLeft": 16,
              "opacity": 1,
            }
          }
        >
          <SvgMock
            height={24}
            style={
              Object {
                "fill": "#333333",
              }
            }
            width={24}
          />
          <Text
            style={
              Array [
                Object {
                  "alignSelf": "center",
                  "color": "#333333",
                  "flexBasis": 0,
                  "flexGrow": 1,
                  "flexShrink": 1,
                  "marginLeft": 32,
                },
              ]
            }
            type="default"
          >
            outlined-finance-moneypaper
          </Text>
        </View>
        <View
          accessible={true}
          focusable={true}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "height": 48,
              "justifyContent": "space-between",
              "marginLeft": 16,
              "opacity": 1,
            }
          }
        >
          <SvgMock
            height={24}
            style={
              Object {
                "fill": "#333333",
              }
            }
            width={24}
          />
          <Text
            style={
              Array [
                Object {
                  "alignSelf": "center",
                  "color": "#333333",
                  "flexBasis": 0,
                  "flexGrow": 1,
                  "flexShrink": 1,
                  "marginLeft": 32,
                },
              ]
            }
            type="default"
          >
            outlined-communication-speaker
          </Text>
        </View>
        <View
          accessible={true}
          focusable={true}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "height": 48,
              "justifyContent": "space-between",
              "marginLeft": 16,
              "opacity": 1,
            }
          }
        >
          <SvgMock
            height={24}
            style={
              Object {
                "fill": "#333333",
              }
            }
            width={24}
          />
          <Text
            style={
              Array [
                Object {
                  "alignSelf": "center",
                  "color": "#333333",
                  "flexBasis": 0,
                  "flexGrow": 1,
                  "flexShrink": 1,
                  "marginLeft": 32,
                },
              ]
            }
            type="default"
          >
            outlined-place-bus
          </Text>
        </View>
        <View
          accessible={true}
          focusable={true}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "height": 48,
              "justifyContent": "space-between",
              "marginLeft": 16,
              "opacity": 1,
            }
          }
        >
          <SvgMock
            height={24}
            style={
              Object {
                "fill": "#333333",
              }
            }
            width={24}
          />
          <Text
            style={
              Array [
                Object {
                  "alignSelf": "center",
                  "color": "#333333",
                  "flexBasis": 0,
                  "flexGrow": 1,
                  "flexShrink": 1,
                  "marginLeft": 32,
                },
              ]
            }
            type="default"
          >
            outlined-alert-info
          </Text>
        </View>
        <View
          accessible={true}
          focusable={true}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "height": 48,
              "justifyContent": "space-between",
              "marginLeft": 16,
              "opacity": 1,
            }
          }
        >
          <SvgMock
            height={24}
            style={
              Object {
                "fill": "#333333",
              }
            }
            width={24}
          />
          <Text
            style={
              Array [
                Object {
                  "alignSelf": "center",
                  "color": "#333333",
                  "flexBasis": 0,
                  "flexGrow": 1,
                  "flexShrink": 1,
                  "marginLeft": 32,
                },
              ]
            }
            type="default"
          >
            outlined-navigation-exit
          </Text>
        </View>
        <Text>
          With External Icon URL
        </Text>
        <View
          accessible={true}
          focusable={true}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "height": 48,
              "justifyContent": "space-between",
              "marginLeft": 16,
              "opacity": 1,
            }
          }
        >
          <Image
            source={
              Object {
                "uri": "https://s3-sa-east-1.amazonaws.com/somos-natura-prd/files/menus/261/img_url_tmp_261_20180323135427.png",
              }
            }
            style={
              Object {
                "height": 24,
                "width": 24,
              }
            }
          />
          <Text
            style={
              Array [
                Object {
                  "alignSelf": "center",
                  "color": "#333333",
                  "flexBasis": 0,
                  "flexGrow": 1,
                  "flexShrink": 1,
                  "marginLeft": 32,
                },
              ]
            }
            type="default"
          >
            Menu Food
          </Text>
        </View>
        <View
          accessible={true}
          focusable={true}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "height": 48,
              "justifyContent": "space-between",
              "marginLeft": 16,
              "opacity": 1,
            }
          }
        >
          <Image
            source={
              Object {
                "uri": "https://s3-sa-east-1.amazonaws.com/somos-natura-prd/files/menus/generic_menu/img_url_tmp_generic_menu_20170822160140.png",
              }
            }
            style={
              Object {
                "height": 24,
                "width": 24,
              }
            }
          />
          <Text
            style={
              Array [
                Object {
                  "alignSelf": "center",
                  "color": "#333333",
                  "flexBasis": 0,
                  "flexGrow": 1,
                  "flexShrink": 1,
                  "marginLeft": 32,
                },
              ]
            }
            type="default"
          >
            Bus Information
          </Text>
        </View>
      </View>
    </RCTScrollView>
  </View>
</RCTSafeAreaView>
`;
